js/src/ui/TxHash/txHash.js

```diff
@@ -87,16 +90,28 @@ class TxHash extends Component {
     const { maxConfirmations } = this.props;
     const { blockNumber, transaction } = this.state;
 
-    let txBlock = 'Pending';
-    let confirmations = 'No';
-    let value = 0;
+    if (!(transaction && transaction.blockNumber && transaction.blockNumber.gt(0))) {
+      return (
+        <div className={ styles.confirm }>
+          <LinearProgress
+            className={ styles.progressbar }
+            color='white'
+            mode='indeterminate'
+          />
+          <div className={ styles.progressinfo }>waiting for confirmations</div>
+        </div>
+      );
+    }
 
-    if (transaction && transaction.blockNumber && transaction.blockNumber.gt(0)) {
-      const num = blockNumber.minus(transaction.blockNumber).plus(1);
-      txBlock = `#${transaction.blockNumber.toFormat(0)}`;
-      confirmations = num.toFormat(0);
-      value = num.gt(maxConfirmations) ? maxConfirmations : num.toNumber();
+    const confirmations = blockNumber.minus(transaction.blockNumber).plus(1);
+    const value = Math.min(confirmations.toNumber(), maxConfirmations);
+    let count;
+    if (confirmations.gt(maxConfirmations)) {
+      count = confirmations.toFormat(0);
+    } else {
+      count = confirmations.toFormat(0) + '/' + maxConfirmations;
```

Comment:
You will always be able to argue with the point of consistency and I agree that it is important. Readable code is more important though imo.

I find

```js
foo + '/' + bar
```

a lot less noisy than

```js
`${a}/${b}`
```

Will adapt to whatever you decide, but keep this in mind.